spring-integration-stream/src/test/java/org/springframework/integration/stream/CharacterStreamSourceTests.java

```diff
@@ -40,4 +46,17 @@ public void testEndOfStream() {
 		assertNull(message2);
 	}
 
+	@Test
+	public void testEOF() {
+		StringReader reader = new StringReader("test");
+		CharacterStreamReadingMessageSource source = new CharacterStreamReadingMessageSource(reader, -1, true);
+		ApplicationEventPublisher publisher = mock(ApplicationEventPublisher.class);
+		source.setApplicationEventPublisher(publisher);
+		Message<?> message1 = source.receive();
+		assertEquals("test", message1.getPayload());
+		Message<?> message2 = source.receive();
+		assertNull(message2);
+		verify(publisher).publishEvent(any(ApplicationEvent.class));
```

Comment:
See the mockito java docs - it doesn't check the type, athough it may do in future.

I'll change it.